### ISSUE TITLE
Purchasing a voucher fails with Integrity constraint violation

### ISSUE THREAD

**Issue by mijewe** [verbb/gift-voucher#40]
### Description
When trying to purchase a voucher (both through the front end and via the CP), the purchase fails with an "Integrity constraint violation" relating to my Voucher Type Layout.

```
SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`local_db`.`elements`, CONSTRAINT `elements_fieldLayoutId_fk` FOREIGN KEY (`fieldLayoutId`) REFERENCES `fieldlayouts` (`id`) ON DELETE SET NULL)
The SQL being executed was: INSERT INTO `elements` (`type`, `uid`, `draftId`, `revisionId`, `fieldLayoutId`, `enabled`, `archived`, `dateCreated`, `dateUpdated`) VALUES ('verbb\\giftvoucher\\elements\\Code', 'd6c7948f-96d1-400e-92d2-3bf1751be765', NULL, NULL, 51, 1, 0, '2020-06-16 15:00:42', '2020-06-16 15:00:42')
```

### Steps to reproduce

1. Create a Voucher Type with some custom layout fields
2. Admin > Commerce > Orders > New Order > Create Your First Line Item > [choose a voucher] > Mark as Completed.
3. You'll see an exception being thrown.

### Additional info
- I have tried deleting my existing voucher types and making them again to see if that kicks the database into shape. It didn't have any effect.

- The payment goes successfully into Commerce, but the order is not marked as complete (and therefore the payment flow doesn't redirect correctly)
- The payment goes successfully into Stripe

- Plugin version: 3.2.1
- Craft version: Craft Pro 3.4.23

**Comment by engram-design** [comment]
Just wanted to confirm you're on [2.3.1](https://github.com/verbb/gift-voucher/releases/tag/2.3.1)?

Are you using a project.yaml file? If so, can you post the contents of `giftVoucher:`? I would also try re-saving Gift Voucher's settings.

**Comment by mijewe** [comment]
Yep - I'm on 2.3.1

Also yes, using the yaml file.

```
giftVoucher:
  voucherTypes:
    db9a8dd5-3b42-47fa-a74a-e1437299e97e:
      handle: giftVouchers
      name: 'Gift Vouchers'
      siteSettings:
        128a4dc0-ab4a-41d4-b5bb-cc54ee19fb9c:
          hasUrls: true
          template: site/pages/gift-voucher/voucher
          uriFormat: 'gift-vouchers/{slug}'
        34b27675-36e2-443b-a7e4-fd062c2c6828:
          hasUrls: true
          template: site/pages/gift-voucher/voucher
          uriFormat: 'gift-vouchers/{slug}'
      skuFormat: 'gift-voucher-{slug}'
      voucherFieldLayouts:
        4039d5af-dd48-47c3-bf48-261b2bdeab14:
          tabs:
            -
              fields:
                8476b5dd-5116-4512-89e0-c93083623b73:
                  required: false
                  sortOrder: 1
                a9dfd71e-a406-4526-a556-59851c222a9a:
                  required: false
                  sortOrder: 2
              name: Content
              sortOrder: 1
```

Resaving the Gift Voucher Settings kicked the yaml into shape, adding
```
  codes:
    fieldLayouts:
      802d2d4f-5eec-4614-9528-96b1bc3b9c7f: null
```
which wasn't there before.
That sorted the issue locally, but not when deploying to another environment with `allowAdminChanges => false`.

I set `allowAdminChanges => true` on my staging environment, resaved the Gift Voucher Settings, and downloaded the yaml into my local environment. The only difference was the UID in the fieldLayouts snippet above.
Saved locally, deployed, and now it works!

---

Not sure if you want to leave this open to investigate the whys, but for me a resave-redeploy sorted this issue out. Thanks!

**Comment by engram-design** [comment]
Thanks for the info - that helps! Should be fixed in [2.4.1](https://github.com/verbb/gift-voucher/releases/tag/2.4.1)
